src/LanguageServer/Test/ImportsTests.cs

```diff
@@ -0,0 +1,301 @@
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Python.Analysis;
+using Microsoft.Python.Analysis.Documents;
+using Microsoft.Python.Core.Text;
+using Microsoft.Python.LanguageServer.Completion;
+using Microsoft.Python.LanguageServer.Sources;
+using Microsoft.Python.LanguageServer.Tests.FluentAssertions;
+using Microsoft.Python.Parsing.Tests;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using TestUtilities;
+
+namespace Microsoft.Python.LanguageServer.Tests {
+    [TestClass]
+    public class ImportsTests : LanguageServerTestBase {
+        public TestContext TestContext { get; set; }
+
+        [TestInitialize]
+        public void TestInitialize()
+            => TestEnvironmentImpl.TestInitialize($"{TestContext.FullyQualifiedTestClassName}.{TestContext.TestName}");
+
+        [TestCleanup]
+        public void Cleanup() => TestEnvironmentImpl.TestCleanup();
+
+        [TestMethod, Priority(0)]
+        public async Task ExplicitImplicitPackageMix() {
+            const string appCode = @"
+import projectA.foo
+import projectA.foo.bar
+import projectB.foo
+import projectB.foo.baz
+
+projectA.";
+
+            var appPath = TestData.GetTestSpecificPath("app.py");
+            var root = Path.GetDirectoryName(appPath);
+            var init1Path = Path.Combine(root, "projectA", "foo", "bar", "__init__.py");
+            var init2Path = Path.Combine(root, "projectA", "foo", "__init__.py");
+            var init3Path = Path.Combine(root, "projectB", "foo", "bar", "__init__.py");
+            var init4Path = Path.Combine(root, "projectB", "foo", "__init__.py");
+
+            await CreateServicesAsync(PythonVersions.LatestAvailable3X, appPath);
+            var rdt = Services.GetService<IRunningDocumentTable>();
+
+            rdt.OpenDocument(new Uri(init1Path), string.Empty);
+            rdt.OpenDocument(new Uri(init2Path), string.Empty);
+            rdt.OpenDocument(new Uri(init3Path), string.Empty);
+            rdt.OpenDocument(new Uri(init4Path), string.Empty);
+
+            var doc = rdt.OpenDocument(new Uri(appPath), appCode, appPath);
+            var analysis = await doc.GetAnalysisAsync();
+
+            var cs = new CompletionSource(new PlainTextDocumentationSource(), ServerSettings.completion);
+            var comps = await cs.GetCompletionsAsync(analysis, new SourceLocation(7, 10));
+            comps.Should().HaveLabels("foo");
```

Comment:
I know in #540 you said you didn't want to use completion to test these; are you fine with this for now then?

Reply:
Yes, it is simpler this way for now